gensim/models/word2vec.py

```diff
@@ -1610,13 +1612,16 @@ def accuracy(self, questions, restrict_vocab=30000, most_similar=most_similar, u
                     logger.debug("skipping line #%i with OOV words: %s" % (line_no, line.strip()))
                     continue
 
+                original_vocab = self.vocab
+                self.vocab = ok_vocab
                 ignore = set(self.vocab[v].index for v in [a, b, c])  # indexes of words to ignore
                 predicted = None
                 # find the most likely prediction, ignoring OOV words and input words
                 sims = most_similar(self, positive=[b, c], negative=[a], topn=False, restrict_vocab=restrict_vocab)
+                self.vocab = original_vocab
                 for index in matutils.argsort(sims, reverse=True):
```

Comment:
Makes sense! 

But, also makes me realize another thorny weirdness of the case-flattened approach. Not just the target words, but the 3 supplied-analogy words may have multiple-case versions. And the one version of those words that survives case-normalization into `ok_vocab` may in fact (due to the arbitrariness of the dict-iteration order) be a less-frequent variant. (For example, a vector index for 'Usa' might be the one that lands in `ok_vocab` over potentially more frequent variants 'USA' or even 'usa'.)

I *think* a principled solution could be to build the case-flattened `ok_vocab` from `index2word`, clipped to `restrict_vocab`, in reverse-order (so last-seen/most-frequent wins). I think this will even match `compute-accuracy.c`'s behavior, because all of its lookups are linear-probes from the beginning of the loaded word-vectors. Does that seem correct to you?